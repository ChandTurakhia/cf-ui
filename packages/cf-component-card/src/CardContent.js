import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const styles = () => ({
  padding: '1.5rem',

  tablet: {
    display: 'table-cell'
  }
});

const Title = createComponent(
  () => ({
    fontWeight: 400,
    marginBottom: '1rem'
  }),
  'h3'
);

const Footer = createComponent(({ theme }) => ({
  fontSize: '0.8rem',
  color: theme.colorGray,
  lineHeight: '1.3',
  marginTop: '1.5em'
}));

const CardContent = ({ className, title, children, footerMessage }) => (
  <div className={className}>
    <Title>{title}</Title>
    {children}
    {footerMessage && <Footer>{footerMessage}</Footer>}
  </div>
);

CardContent.propTypes = {
  title: PropTypes.node,
  footerMessage: PropTypes.string,
  children: PropTypes.node
};

export default createComponent(styles, CardContent);
