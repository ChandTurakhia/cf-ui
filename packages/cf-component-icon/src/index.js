import IconUnstyled from './Icon';
import IconTheme from './IconTheme';

import { applyTheme } from 'cf-style-container';

const Icon = applyTheme(IconUnstyled, IconTheme);

export { Icon, IconUnstyled, IconTheme };
