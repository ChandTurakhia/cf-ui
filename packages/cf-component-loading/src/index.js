import LoadingUnstyled from './Loading';
import LoadingTheme from './LoadingTheme';

import { applyTheme } from 'cf-style-container';

const Loading = applyTheme(LoadingUnstyled, LoadingTheme);

export { Loading, LoadingUnstyled, LoadingTheme };
