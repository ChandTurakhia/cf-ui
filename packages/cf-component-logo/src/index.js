import Logo from './Logo';
import logoSvg from '../assets/logo.svg';
import logoSvgMobile from '../assets/logo_mobile.svg';

export { Logo, logoSvg, logoSvgMobile };
