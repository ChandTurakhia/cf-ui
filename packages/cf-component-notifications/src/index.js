import NotificationUnstyled from './Notification';
import NotificationListUnstyled from './NotificationList';
import NotificationGlobalContainerUnstyled from './NotificationGlobalContainer';

import NotificationTheme from './NotificationTheme';
import NotificationListTheme from './NotificationListTheme';
import NotificationGlobalContainerTheme
  from './NotificationGlobalContainerTheme';

import { applyTheme } from 'cf-style-container';

const Notification = applyTheme(NotificationUnstyled, NotificationTheme);

const NotificationList = applyTheme(
  NotificationListUnstyled,
  NotificationListTheme
);

const NotificationGlobalContainer = applyTheme(
  NotificationGlobalContainerUnstyled,
  NotificationGlobalContainerTheme
);

export {
  Notification,
  NotificationList,
  NotificationGlobalContainer,
  NotificationUnstyled,
  NotificationListUnstyled,
  NotificationGlobalContainerUnstyled,
  NotificationTheme,
  NotificationListTheme,
  NotificationGlobalContainerTheme
};
