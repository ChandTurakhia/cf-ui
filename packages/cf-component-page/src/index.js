import PageUnstyled from './Page';
import PageHeaderUnstyled from './PageHeader';
import PageContentUnstyled from './PageContent';
import PageTheme from './PageTheme';
import Header from './Header';
import Logo from './Logo';
import NavList from './NavList';
import NavItem from './NavItem';
import logoSvg from './logo.svg';
import logoSvgMobile from './logo_mobile.svg';

import { applyTheme } from 'cf-style-container';

const Page = applyTheme(PageUnstyled, PageTheme);
const PageHeader = applyTheme(PageHeaderUnstyled, {});
const PageContent = applyTheme(PageContentUnstyled, {});

export {
  Header,
  Logo,
  logoSvg,
  logoSvgMobile,
  NavItem,
  NavList,
  Page,
  PageTheme,
  PageUnstyled,
  PageHeader,
  PageHeaderUnstyled,
  PageContent,
  PageContentUnstyled
};
