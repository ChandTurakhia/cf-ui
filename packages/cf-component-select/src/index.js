import { applyTheme } from 'cf-style-container';

import SelectUnstyled from './Select';
import SelectTheme from './SelectTheme';

const Select = applyTheme(SelectUnstyled, SelectTheme);

export { Select, SelectUnstyled, SelectTheme };
