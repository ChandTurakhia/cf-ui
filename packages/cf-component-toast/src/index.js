import { applyTheme } from 'cf-style-container';

import ToastUnstyled from './Toast';
import ToastTheme from './ToastTheme';

const Toast = applyTheme(ToastUnstyled, ToastTheme);

export { Toast, ToastUnstyled, ToastTheme };
