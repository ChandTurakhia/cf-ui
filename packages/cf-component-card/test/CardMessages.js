import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { CardMessages } from '../../cf-component-card/src/index';

test('should render', () => {
  const snapshot = felaSnapshot(
    <CardMessages
      messages={[
        { type: 'info', content: 'Message One' },
        { type: 'error', content: 'Message Two' }
      ]}
    />
  );
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
