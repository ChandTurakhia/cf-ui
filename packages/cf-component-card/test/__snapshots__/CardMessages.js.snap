// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="f113e28i"
>
  <div
    role="alert"
  >
    Message One
  </div>
  <div
    role="alert"
  >
    Message Two
  </div>
</div>
`;

exports[`should render 2`] = `
"
.f113e28i {
  padding-top: 0.5rem
}
"
`;
